rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/transform/TbDeleteKeysNode.java

```diff
@@ -21,84 +21,96 @@
 import org.thingsboard.common.util.JacksonUtil;
 import org.thingsboard.rule.engine.api.RuleNode;
 import org.thingsboard.rule.engine.api.TbContext;
-import org.thingsboard.rule.engine.api.TbNode;
 import org.thingsboard.rule.engine.api.TbNodeConfiguration;
 import org.thingsboard.rule.engine.api.TbNodeException;
 import org.thingsboard.rule.engine.api.util.TbNodeUtils;
+import org.thingsboard.rule.engine.util.TbMsgSource;
 import org.thingsboard.server.common.data.plugin.ComponentType;
 import org.thingsboard.server.common.msg.TbMsg;
 import org.thingsboard.server.common.msg.TbMsgMetaData;
 
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ExecutionException;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 @Slf4j
 @RuleNode(
         type = ComponentType.TRANSFORMATION,
-        name = "delete keys",
+        name = "delete key-values",
+        version = 1,
         configClazz = TbDeleteKeysNodeConfiguration.class,
-        nodeDescription = "Removes keys from the msg data or metadata with the specified key names selected in the list",
-        nodeDetails = "Will fetch fields (regex) values specified in list. If specified field (regex) is not part of msg " +
-                "or metadata fields it will be ignored. Returns transformed messages via <code>Success</code> chain",
+        nodeDescription = "Removes key-values from message or message metadata.",
+        nodeDetails = "Removes key-values from message or message metadata based on the keys list specified in the configuration. " +
+                "Use regular expression(s) as a key(s) to remove keys by pattern.<br><br>" +
+                "Output connections: <code>Success</code>, <code>Failure</code>.",
```

Comment:
This node deletes specified key-value pairs from the message body or metadata. Keys and/or regular expressions specified in configuration are used to determine which keys should be removed.